File: src/main/java/ar/edu/utn/frba/dds/models/domain/comunidades/gradosDeConfianza/Puntaje.java
package ar.edu.utn.frba.dds.models.domain.comunidades.gradosDeConfianza;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter

@Embeddable
public class Puntaje {
    @Setter
    @Column(name = "puntaje")
    private Double valor;

    @Embedded
    private GradoDeConfianza gradoDeConfianza;

    public Puntaje(Double valor){
        this.valor = valor;
        this.gradoDeConfianza = new GradoDeConfianza();
    }

    public Puntaje() {

    }


    public void actualizarPuntaje(double puntajeNuevo){

        this.valor = puntajeNuevo;
        calcularGrado(puntajeNuevo);
    }

    private void calcularGrado(double puntajeNuevo) {
        if(puntajeNuevo < 2){
            this.gradoDeConfianza.setNombre(TipoDeGrado.NO_CONFIABLE);
        } else if (puntajeNuevo>=2 && puntajeNuevo<=3){
            this.gradoDeConfianza.setNombre(TipoDeGrado.CON_RESERVAS);
        } else {
            this.gradoDeConfianza.setNombre(TipoDeGrado.CONFIABLE);
            this.gradoDeConfianza.calcularNivel(puntajeNuevo);
        }
    }




}

File: src/main/ts/ar/edu/utn/frba/dds/domain/comunidades/gradosDeConfianza/Puntaje.ts
import {GradoDeConfianza} from "./GradoDeConfianza";
import {TipoDeGrado} from "./TipoDeGrado";

export class Puntaje {
    _valor: number;
    gradoDeConfianza: GradoDeConfianza;

    constructor () {
        this.gradoDeConfianza = new GradoDeConfianza();
    }


    get valor () : number {
        return this._valor;
    }

    set valor (puntajeNuevo) {
        this._valor = puntajeNuevo;
        this.calcularGrado (puntajeNuevo);
    }

    calcularGrado (puntaje: number) {
        if (puntaje < 2) {
            this.gradoDeConfianza.nombre = TipoDeGrado.NO_CONFIABLE;
        }
        else if (3 >= puntaje && puntaje >= 2) {
            this.gradoDeConfianza.nombre = TipoDeGrado.CON_RESERVAS;
        }
        else {
            this.gradoDeConfianza.nombre = TipoDeGrado.CONFIABLE;
            this.gradoDeConfianza.calcularNivel(puntaje);
        }
    }
}